src/uct/sm/mm/base/mm_md.c

```diff
@@ -67,25 +67,26 @@ ucs_status_t uct_mm_seg_new(void *address, size_t length, uct_mm_seg_t **seg_p)
     return UCS_OK;
 }
 
-void uct_mm_md_query(uct_md_h md, uct_md_attr_t *md_attr, uint64_t max_alloc)
+void uct_mm_md_query(uct_md_h md, uct_md_attr_t *md_attr, size_t max_alloc,
+                     size_t max_reg)
 {
-    memset(md_attr, 0, sizeof(*md_attr));
-
     md_attr->cap.flags            = UCT_MD_FLAG_RKEY_PTR |
                                     UCT_MD_FLAG_NEED_RKEY;
-    md_attr->cap.max_reg          = 0;
-    md_attr->cap.max_alloc        = 0;
+    md_attr->cap.reg_mem_types    = 0;                                
     md_attr->cap.alloc_mem_types  = UCS_BIT(UCS_MEMORY_TYPE_HOST);
     md_attr->cap.cache_mem_types  = UCS_BIT(UCS_MEMORY_TYPE_HOST);
     md_attr->cap.access_mem_types = UCS_BIT(UCS_MEMORY_TYPE_HOST);
     md_attr->cap.detect_mem_types = 0;
 
     if (max_alloc > 0) {
-        md_attr->cap.flags       |= UCT_MD_FLAG_ALLOC | UCT_MD_FLAG_FIXED;
-        md_attr->cap.max_alloc    = max_alloc;
+        md_attr->cap.flags    |= UCT_MD_FLAG_ALLOC | UCT_MD_FLAG_FIXED;
+        md_attr->cap.max_alloc = max_alloc;
     }
 
-    memset(&md_attr->local_cpus, 0xff, sizeof(md_attr->local_cpus));
+    if (max_reg > 0) {
+        md_attr->cap.flags  |= UCT_MD_FLAG_REG;
+        md_attr->cap.max_reg = max_reg;
+    }
```

Comment:
why needed here? Can just leave it in xpmem

Reply:
it is moved to the common code

Reply:
why needed, if it is relevant for xpmem only?

Reply:
I see, removed